### PR TITLE
added echarts

### LINKED ISSUE #779: Use the bazaar endpoints for bazaar items
Requires #777 
The item details endpoint returns if an item is on bazaar (via flags)
if its on bazaar it should display a special ui taking advantage of the time series format of the endpoints.

Docs for flags
        //
        // Summary:
        //     Enum NONE for value: NONE
        NONE = 0,
        //
        // Summary:
        //     Enum BAZAAR for value: BAZAAR
        BAZAAR = 1,
        //
        // Summary:
        //     Enum TRADEABLE for value: TRADEABLE
        TRADEABLE = 2,
        //
        // Summary:
        //     Enum AUCTION for value: AUCTION
        AUCTION = 4,
        //
        // Summary:
        //     Enum CRAFT for value: CRAFT
        CRAFT = 8,
        //
        // Summary:
        //     Enum GLOWING for value: GLOWING
        GLOWING = 16,
        //
        // Summary:
        //     Enum MUSEUM for value: MUSEUM
        MUSEUM = 32

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #779

### CODE REVIEW

File: components/PriceGraph/AuctionHousePriceGraph/AuctionHousePriceGraph.tsx [hunk 1]
```diff
@@ -163,6 +155,22 @@ function PriceGraph(props: Props) {
...
+    function setSelectedLegendOptionsFromLocalStorage() {
+        let legendSelected = JSON.parse(localStorage.getItem(AUCTION_GRAPH_LEGEND_SELECTION) || '{}')
+        chartOptions.legend.selected = legendSelected || chartOptions.legend.selected
+        setChartOptions(chartOptions)
+    }
+
+    function onChartsEvents(): Record<string, Function> {
+        return {
+            legendselectchanged: e => {
+                let current = JSON.parse(localStorage.getItem(AUCTION_GRAPH_LEGEND_SELECTION) || '{}')
```
Comment (Ekwav) [inline components/PriceGraph/AuctionHousePriceGraph/AuctionHousePriceGraph.tsx]: this is the same as in line 159
Please refactor it into a function

File: components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+                disableAllTime={false}
+                item={props.item}
+            />
+
+            <div>
+                <div className={styles.chartsWrapper}>
+                    {!isSSR ? (
+                        <>
+                            <div className={graphType === GRAPH_TYPE.SINGLE ? styles.chartWrapperSingle : styles.chartWrapperSplit}>
+                                <h3 className={styles.graphHeadline}>{graphType === GRAPH_TYPE.SINGLE ? 'Buy data' : 'Bazaar data'}</h3>
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx]: this if is the wrong way around

File: components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+}
+
+const DEFAULT_GRAPH_TYPE = GRAPH_TYPE.SPLIT
+
+let currentLoadingString
+
+// Boolean if the component is mounted. Set to false in useEffect cleanup function
+let mounted = true
+
+function BazaarPriceGraph(props: Props) {
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx:36]: styling for mobile only uses half the space
image

File: components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+                            >
+                                <option value={GRAPH_TYPE.SINGLE}>Single</option>
+                                <option value={GRAPH_TYPE.SPLIT}>Split</option>
+                            </Form.Control>
+                        ) : null}
+                    </div>
+                    <div style={{ float: 'right' }}>
+                        <ShareButton
+                            title={'Prices for ' + props.item.name}
+                            text="See list, search and filter item prices from the auction house and bazar in Hypixel Skyblock"
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx]:
```suggestion
                            text="Browse the Bazaar history in Hypixel Skyblock"
```

File: components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                crossStyle: {
+                    color: '#999'
+                },
+                label: {
+                    formatter: axisObject => {
+                        if (axisObject.axisDimension === 'y') {
+                            return `${numberWithThousandsSeperators(axisObject.value)}`
+                        }
+                        let d = new Date(+axisObject.value)
+                        return d.toLocaleTimeString() + ', ' + d.toLocaleDateString()
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx]: why is the time first? usually the date is first and the time is second

File: components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+                type: 'line',
+                color: '#4B0082',
+                smooth: true,
+                symbol: 'none',
+                yAxisIndex: 1,
+                tooltip: {
+                    show: true,
+                    valueFormatter: value => `${numberWithThousandsSeperators(value)}`
+                },
+                data: []
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx]: this is identical with the settings above and below, why isn't this a variable of some sorts?
maybe just a template expanded with the ... operator

File: components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSplit.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                type: 'line',
+                color: '#4B0082',
+                yAxisIndex: 1,
+                smooth: true,
+                symbol: 'none',
+                tooltip: {
+                    show: true,
+                    valueFormatter: value => `${numberWithThousandsSeperators(value)}`
+                },
+                data: []
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSplit.tsx]: repetitive again, please avoid copy pasting

File: package.json [hunk 8]
```diff
@@ -16,6 +16,7 @@
     "bootstrap-icons": "^1.7.2",
     "chart.js": "^2.9.3",
```
Comment (Ekwav) [inline package.json]: why is this still here, didn't it get full replaced by echarts?

**Review by Ekwav — CHANGES_REQUESTED** [review]
also I am wondering where the notification feature went for bazaar

File: components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx (hunk 6 above)
Comment (matthias-luger) [reply components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx]: Dont really know what you mean with a template being extended with `...`, but I think I refactored it good enough :)

File: components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx (hunk 3 above)
Comment (matthias-luger) [reply components/PriceGraph/BazaarPriceGraph/BazaarPriceGraph.tsx:36]: image
image

Cant reproduce :(

**Comment by matthias-luger** [comment]
> also I am wondering where the notification feature went for bazaar

I think I should add/rework that in a new issue (this one is kinda big enough).
In the last implementation it was the same as the auction notification which doesnt really make sense for the most part.
image

File: components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+                }
+            },
+            {
+                name: 'Min (buy)',
+                type: 'line',
+                color: '#228B22',
+                smooth: true,
+                symbol: 'none',
+                yAxisIndex: 0,
+                animationThreshold: 200,
```
Comment (Ekwav) [inline components/PriceGraph/BazaarPriceGraph/PriceGraphConfigSingle.tsx]: redundant, please remove redundancy

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Comment by Ekwav** [comment]
The config is still redundant but apparently you aren't interested in changing that

**Pull request merged**